### PR TITLE
refactor: Remove error returns from config functions, fix tests.

### PR DESCRIPTION
**Pull request by kentquirk**
## Which problem is this PR solving?

- Since the introduction of validation of config, the config functions can't fail, and the error returns are useless. 

## Short description of the changes

- Remove error returns from all the config getters
- Fix all the places we used (or didn't use them)
- Fix a couple of tests that assumed things that are no longer true

### CODE REVIEW

File: test/config.yaml [hunk 1]
```diff
@@ -13,13 +13,11 @@ Logger:
 LegacyMetrics:
   Enabled: true
   Dataset: refinery_metrics
-  APIKey: $REFINERY_HONEYCOMB_API_KEY
```
Comment (VinozzZ) [inline test/config.yaml:16]: Why are these deleted?

**Review by VinozzZ — APPROVED** [review]
Yahoo!

File: test/config.yaml (hunk 1 above)
Comment (kentquirk) [reply test/config.yaml:16]: Because the $ syntax doesn't work, but there are environment variables that are automatic if they're set.
